packages/p2p/src/components/my-ads/my-ads.jsx

```diff
@@ -14,63 +15,56 @@ const MyAdsState = ({ message }) => (
     </div>
 );
 
-class MyAds extends React.Component {
-    state = {
-        show_form: false,
-    };
-
-    handleShowForm = show_form => {
-        this.setState({ show_form });
-    };
-
-    componentDidMount() {
-        this.is_mounted = true;
+const MyAds = () => {
+    const { is_advertiser, is_restricted, setPoiStatus } = React.useContext(Dp2pContext);
+    const [is_loading, setIsLoading] = React.useState(true);
+    const [show_ad_form, setShowAdForm] = React.useState(false);
 
-        if (!this.context.is_advertiser) {
+    React.useEffect(() => {
+        if (!is_advertiser) {
             requestWS({ get_account_status: 1 }).then(response => {
-                if (this.is_mounted && !response.error) {
+                if (!response.error) {
```

Comment:
Since this is async we should add a check for whether this component is mounted before updating any of its state (`setIsLoading`).